### PR TITLE
Feature/onboarding

### PR DESCRIPTION
**Pull request by avneesh-akto**

### CODE REVIEW

File: apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java [hunk 1]
```diff
@@ -1,52 +1,108 @@
...
+        String victimToken = tokens.get("victimToken");
+        harString = harString.replaceAll("\\{\\{AKTO\\.token\\}\\}", victimToken);
+
+        // process har file
+        HarAction harAction = new HarAction();
+        harAction.setHarString(harString);
+        harAction.setApiCollectionName("juice_shop_demo");
+        Map<String, Object> session = new HashMap<>();
+        harAction.setSession(session);
+        // todo: skipKafka = true for onPrem also
```
Comment (aktoboy) [inline apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java:80]: Do we need to handle this TODO?

File: apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java [hunk 2]
```diff
@@ -1,52 +1,108 @@
...
+
+        // auth mechanism
+        String attackerKey = tokens.get("attackerKey");
+        String attackerToken  = tokens.get("attackerToken");
+        List<AuthParam> authParamList = new ArrayList<>();
+        authParamList.add(new HardcodedAuthParam(AuthParam.Location.HEADER, attackerKey, attackerToken, true));
+        AuthMechanism authMechanism = new AuthMechanism(
+             authParamList, new ArrayList<>(), "HARDCODED"
+        );
+        AuthMechanismsDao.instance.insertOne(authMechanism);
```
Comment (aktoboy) [inline apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java:95]: Once the demo collection's testing is executed, we will also need to delete these tokens, right?

File: apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java [hunk 3]
```diff
@@ -1,52 +1,108 @@
...
-            }
-        }, 0 , TimeUnit.SECONDS);
+    public void runMainFunction() {
+        Context.accountId.set(1_000_000);
+        Main.initializeRuntime();
+        httpCallParser = new HttpCallParser("userIdentifier", 1, 1, 1, false);
+        aktoPolicy = new AktoPolicy(RuntimeListener.httpCallParser.apiCatalogSync, false);
 
+        try {
+            initialiseDemoCollections();
```
Comment (aktoboy) [inline apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java:39]: What will happen to user onboarding if there is an exception here?

File: apps/dashboard/web/src/apps/dashboard/views/testing/components/TestingRunResults.vue [hunk 4]
```diff
@@ -169,7 +185,21 @@ export default {
             isWorkflow: false,
             originalStateFromDb: null,
             dialogBoxIssue: {},
-            similarlyAffectedIssues: []
+            similarlyAffectedIssues: [],
+            loading: true,
+            jokes: [
+                ["Looking for love in all the wrong places", "../../etc/pwd"],
```
Comment (aktoboy) [inline apps/dashboard/web/src/apps/dashboard/views/testing/components/TestingRunResults.vue:191]: Jokes were supposed to be maintained in a seperate file, right?

File: apps/dashboard/web/src/apps/dashboard/views/testing/components/TestingRunResults.vue [hunk 5]
```diff
@@ -231,7 +261,7 @@ export default {
             this.selectedDate = point / 1000
         },
         refreshSummaries() {
-            api.fetchTestingRunResultSummaries(this.startTimestamp, this.endTimestamp, this.testingRunHexId).then(resp => {
+            return api.fetchTestingRunResultSummaries(this.startTimestamp, this.endTimestamp, this.testingRunHexId).then(resp => {
```
Comment (aktoboy) [inline apps/dashboard/web/src/apps/dashboard/views/testing/components/TestingRunResults.vue:264]: typo?

**Review by aktoboy — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java (hunk 1 above)
Comment (avneesh-akto) [reply apps/dashboard/src/main/java/com/akto/listener/RuntimeListener.java:80]: no

**Pull request merged**